src/Microsoft.AspNet.Mvc.Xml/RequiredValidationHelper.cs

```diff
@@ -0,0 +1,97 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Reflection;
+using System.Runtime.Serialization;
+using Microsoft.AspNet.Mvc.ModelBinding;
+
+namespace Microsoft.AspNet.Mvc.Xml
+{
+    public static class RequiredValidationHelper
+    {
+        public static void Validate(Type modelType, ModelStateDictionary modelStateDictionary)
+        {
+            Validate(modelType, modelStateDictionary, new HashSet<Type>());
```

Comment:
FYI..I wanted to use the `ReferenceEqualityComparer` from ModelBinding but it is internal to it